gonzo/clouds/compute.py

```diff
@@ -0,0 +1,254 @@
+        return instance_types
+
+    def list_instance_tags(self, node):
+        return node.extra[self.TAG_KEY]
+
+    def list_availability_zones(self):
+        return self.compute_session.list_locations()
+
+    def get_next_az(self, server_type):
+        availabie_azs = self.list_availability_zones()
+        try:
+            newest_instance_az = self.list_instances_by_type(
+                server_type)[-1].extra['gonzo_az']
+        except IndexError:
+            print "No Server Types Found"
+            return availabie_azs[0]
+
+        if len(availabie_azs) == 1:
+            return availabie_azs[0]
+        else:
+            for index, availabie_az in enumerate(availabie_azs):
+                if availabie_az.name == newest_instance_az:
+                    if (index + 1) == len(availabie_azs):
+                        return availabie_azs[0]
+                    else:
+                        return availabie_azs[index + 1]
+
+    def create_instance(self, image_name, name, owner, user_data=None,
+                        security_groups=None, size=None, key_name=None):
+        instance_name = name.split('-')
+        server_type = '-'.join(instance_name[1:][:-1])
+
+        # Image
+        image = self.get_image(image_name)
+
+        # Key Pair
+        if key_name is not None:
+            key = self.get_key_pair(key_name)
+            key_name = key.name
+
+        # Tags
+        tags = self.generate_instance_metadata(
+            owner,
+            environment=instance_name[0],
+            server_type=server_type
+            )
+
+        # Instance Size
+        size = self.get_instance_size(size, self.INSTANCE_SIZE_ATTRIBUTE)
+
+        # Security Groups
+        if security_groups is None:
+            security_groups = []
+
+        sec_group_objects = []
+
+        for security_group in security_groups:
+            self.create_if_not_exist_security_group(security_group)
+            sec_group_objects.append(self.get_security_group(security_group))
+
+        if not self.SECURITY_GROUP_OBJECT:
+            security_groups = []
+            for sec_group in sec_group_objects:
+                security_groups.append(getattr(sec_group,
+                                               self.SECURITY_GROUP_IDENTIFIER))
+        else:
+            security_groups = sec_group_objects
+
+        # Availability Zone
+        az = self.get_next_az(server_type)
+
+        # Launch Instance
+        instance = self.compute_session.create_node(
+            name=name,
+            image=image,
+            size=size,
+            location=az,
+            ex_security_groups=security_groups,
+            ex_metadata=tags,
+            ex_userdata=user_data,
+            ex_keyname=key_name,
+            )
+        self.compute_session.wait_until_running([instance])
+        new_instance = self.get_instance_by_uuid(instance.uuid)
+        return new_instance
+
+    def get_instance_size(self, size_name, query_attribute):
```

Comment:
are these method in any kind of order? could we group/sort them in some sensible way?